Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,276 @@
+                        TagId = tagToAdd.Id
+                    };
+                    context.EventTags.Add(eventTag);
+                    context.SaveChanges();
+                }
+                else
+                {
+                    var eventTag = new EventTag
+                    {
+                        EventId = eventId,
+                        TagId = newTag.Id
+                    };
+                    context.EventTags.Add(eventTag);
+                    context.SaveChanges();
+                }
+            }
+        }
+
+        return eventDto;
+    }
+
+    public EventDto UpdateEvent(int eventId, EventDto eventDto)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var eventToUpdate = context.Events.FirstOrDefault(r => r.Id == eventId);
+            if (eventToUpdate == null)
+            {
+                return null;
+            }
+
+            eventToUpdate.Name = eventDto.Name;
+            eventToUpdate.AddressId = eventDto.AddressId;
+            eventToUpdate.Description = eventDto.Description;
+            eventToUpdate.OrganizerId = eventDto.OrganizerId;
+            eventToUpdate.StartAt = eventDto.StartAt;
+            eventToUpdate.EndAt = eventDto.EndAt;
+            eventToUpdate.UpdatedAt = DateTime.Now;
+            eventToUpdate.StatusId = eventDto.StatusId;
+
+            context.Events.Update(eventToUpdate);
+            context.SaveChanges();
+
+            var existingTags = context.EventTags.Where(r => r.EventId == eventId).Include(eventTag => eventTag.Tag).ToList();
+
+            if (eventDto.Tags == null)
+            {
+                return eventDto;
+            }
+
+            var newTags = eventDto.Tags
+                .Select(tagDto => new Tag
+                {
+                    Name = tagDto.Name,
+                }).ToList();
+
+            var tagsToRemove = from existingTag in existingTags
+                               where newTags.All(r => r.Name != existingTag.Tag.Name)
+                               select existingTag;
+
+            foreach (var existingTag in tagsToRemove)
+            {
+                context.EventTags.Remove(existingTag);
+            }
+
+            var tagsToAdd = from newTag in newTags
+                            where existingTags.All(r => r.Tag.Name != newTag.Name)
+                            select newTag;
+
+            foreach (var newTag in tagsToAdd)
+            {
+                var existingTag = context.Tags.FirstOrDefault(t => t.Name == newTag.Name);
+                if (existingTag == null)
+                {
+                    existingTag = new Tag { Name = newTag.Name };
+                    context.Tags.Add(existingTag);
+                    context.SaveChanges();
+                }
+
+                context.EventTags.Add(new EventTag { EventId = eventId, TagId = existingTag.Id });
+            }
+
+            context.SaveChanges();
+        }

```

Comment:
We can from tagsToAdd get those Tags which are not on the base and return directly new Tag { Name = tag.Name } for each of them, then use .AddRange for Tags and via linq retrieve EventTags which are needed to be added and do similar, AddRange for them.